bindings/c/box.cpp

```diff
@@ -0,0 +1,97 @@
+#include <manifold.h>
+#include <public.h>
+
+#include <algorithm>
+#include <conv.hh>
+#include <vector>
+
+#include "types.h"
+
+ManifoldBox *manifold_box(void *mem, float x1, float y1, float z1, float x2,
+                          float y2, float z2) {
+  auto p1 = glm::vec3(x1, y1, z1);
+  auto p2 = glm::vec3(x2, y2, z2);
+  auto box = new (mem) Box(p1, p2);
```

Comment:
What does this syntax mean? Is it just constructing a box at the pointer location `mem` and assuming there's enough space allocated there?

Reply:
Yeap that's right